sky/clouds/kubernetes.py

```diff
@@ -0,0 +1,364 @@
+        # exactly the requested resources.
+        instance_cpus = int(
+            cpus.strip('+')) if cpus is not None else cls._DEFAULT_NUM_VCPUS
+        instance_mem = int(
+            memory.strip('+')
+        ) if memory is not None else \
+            instance_cpus * cls._DEFAULT_MEMORY_CPU_RATIO
+        virtual_instance_type = KubernetesInstanceType(instance_cpus,
+                                                       instance_mem).name
+        return virtual_instance_type
+
+    @classmethod
+    def get_accelerators_from_instance_type(
+        cls,
+        instance_type: str,
+    ) -> Optional[Dict[str, int]]:
+        # TODO(romilb): Add GPU support.
+        return None
+
+    @classmethod
+    def get_vcpus_mem_from_instance_type(
+            cls, instance_type: str) -> Tuple[Optional[float], Optional[float]]:
+        """Returns the #vCPUs and memory that the instance type offers."""
+        k = KubernetesInstanceType.from_instance_type(instance_type)
+        return k.cpus, k.memory
+
+    @classmethod
+    def zones_provision_loop(
+        cls,
+        *,
+        region: str,
+        num_nodes: int,
+        instance_type: str,
+        accelerators: Optional[Dict[str, int]] = None,
+        use_spot: bool = False,
+    ) -> Iterator[Optional[List[clouds.Zone]]]:
+        del num_nodes, region, instance_type, accelerators, use_spot  # Unused.
+        for r in cls.regions():
+            yield r.zones
+
+    @classmethod
+    def get_zone_shell_cmd(cls) -> Optional[str]:
+        return None
+
+    def make_deploy_resources_variables(
+            self, resources: 'resources_lib.Resources',
+            region: Optional['clouds.Region'],
+            zones: Optional[List['clouds.Zone']]) -> Dict[str, Optional[str]]:
+        del zones
+        if region is None:
+            region = self._regions[0]
+
+        r = resources
+        acc_dict = self.get_accelerators_from_instance_type(r.instance_type)
+        if acc_dict is not None:
+            custom_resources = json.dumps(acc_dict, separators=(',', ':'))
+        else:
+            custom_resources = None
+
+        # resources.memory and cpus are None if they are not explicitly set.
+        # We fetch the default values for the instance type in that case.
+        cpus, mem = self.get_vcpus_mem_from_instance_type(
+            resources.instance_type)
+        return {
+            'instance_type': resources.instance_type,
+            'custom_resources': custom_resources,
+            'region': region.name,
+            'cpus': str(cpus),
+            'memory': str(mem),
+            'timeout': str(self.TIMEOUT),
+            'k8s_ssh_key_secret_name': self.SKY_SSH_KEY_SECRET_NAME,
+            # TODO(romilb): Allow user to specify custom images
+            'image_id': self.IMAGE,
+        }
+
+    def get_feasible_launchable_resources(self,
+                                          resources: 'resources_lib.Resources'):
+        if resources.use_spot:
```

Comment:
Since disk tier is a generally a soft-constraint (i.e., job won't fail with a lower disk-tier, would just run slower), I was thinking it's fine to "ignore" disk tier for k8s to enable failover when disk tier is specified. wdyt? (I don't feel too strongly, If we want to be consistent with lambda I can change it here)